evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -164,72 +139,54 @@ pub(crate) fn simulate_cpu_and_get_user_jumps<F: Field>(
     }
 }
 
-/// Different types of Memory operations in the interpreter, and the data
-/// required to revert them.
-enum InterpreterMemOpKind {
-    /// We need to provide the context.
-    Push(usize),
-    /// If we pop a certain value, we need to push it back to the correct
-    /// context when reverting.
-    Pop(U256, usize),
-    /// If we write a value at a certain address, we need to write the old value
-    /// back when reverting.
-    Write(U256, usize, usize, usize),
+/// Different types of Memory operations in the interpreter.
+#[derive(Debug)]
+pub(crate) enum InterpreterMemOpKind {
+    Read(U256, MemoryAddress),
+    Write(U256, MemoryAddress),
 }
 
-impl<'a, F: Field> Interpreter<'a, F> {
-    pub(crate) fn new_with_kernel(initial_offset: usize, initial_stack: Vec<U256>) -> Self {
-        let mut result = Self::new(
-            &KERNEL.code,
-            initial_offset,
-            initial_stack,
-            &KERNEL.prover_inputs,
-        );
-        result.initialize_rlp_segment();
-        result
-    }
-
+impl<F: Field> Interpreter<F> {
     /// Returns an instance of `Interpreter` given `GenerationInputs`, and
     /// assuming we are initializing with the `KERNEL` code.
     pub(crate) fn new_with_generation_inputs_and_kernel(
```

Comment:
Given that we can't really run the interpreter without the kernel, perhaps we can rename this and other methods? The `and_kernel` part seems superfluous?